server/controllers/client.ts

```diff
@@ -63,14 +65,14 @@ const controllers: IControllerClient = {
     const { params, query, sort } = ctx;
     const { relation } = parseParams<{ relation: string }>(params);
 
-    const { sort: querySort, ...filterQuery } = query || {};
+    const { sort: querySort, fields, ...filterQuery } = query || {};
 
     try {
       assertParamsPresent<{ relation: string }>(params, ["relation"]);
 
       return await this.getService<IServiceCommon>("common").findAllInHierarchy(
         {
-          ...flatInput(relation, filterQuery, sort || querySort),
+          ...flatInput<Comment>(relation, filterQuery, sort || querySort, undefined, fields),
```

Comment:
Maybe it would be better to pass arguments as an object rather than remember to place undefined every time. 🤔